### PR TITLE
Changes to use RetryCallback to handle token expiration 

### PR DESCRIPTION
**Pull request by fegabe**
I want to perform a refresh token when expired to do that I need several changes in the library. See discussion here for more details: https://github.com/proyecto26/RestClient/issues/141

### CODE REVIEW

**Comment by jdnichollsc** [comment]
Hello mate, sorry for the delay, can you update your changes to merge this awesome fix?

**Comment by fegabe** [comment]
That should do the trick. Gracias por tu trabajo!

File: src/Proyecto26.RestClient/Helpers/HttpBase.cs [hunk 1]
```diff
@@ -46,15 +46,15 @@ public static IEnumerator CreateRequestAndRetry(RequestHelper options, Action<Re
                         callback(null, response);
                         break;
                     }
-                    else if (!options.IsAborted && retries < options.Retries && (IsNetworkError))
+                    else if (!options.IsAborted && retries < options.Retries && (!options.RetryCallbackOnlyOnNetworkErrors || request.isNetworkError))
```
Comment (jdnichollsc) [inline src/Proyecto26.RestClient/Helpers/HttpBase.cs:49]: Hello mate, can you use `IsNetworkError` instead of `request.isNetworkError` to support new versions of Unity, please?

**Review by jdnichollsc — CHANGES_REQUESTED** [review]
I leave a comment, thanks for your help bro!

File: src/Proyecto26.RestClient/Helpers/HttpBase.cs (hunk 1 above)
Comment (fegabe) [reply src/Proyecto26.RestClient/Helpers/HttpBase.cs]: sure! sorry I didn't realize about it. Now it's changed

File: src/Proyecto26.RestClient/Helpers/HttpBase.cs [hunk 2]
```diff
@@ -46,15 +46,15 @@ public static IEnumerator CreateRequestAndRetry(RequestHelper options, Action<Re
...
                     {
-                        yield return new WaitForSeconds(options.RetrySecondsDelay);
-                        retries++;
                         if (options.RetryCallback != null)
                         {
                             options.RetryCallback(CreateException(options, request), retries);
                         }
                         DebugLog(options.EnableDebug, string.Format("RestClient - Retry Request\nUrl: {0}\nMethod: {1}", options.Uri, options.Method), false);
+                        yield return new WaitForSeconds(options.RetrySecondsDelay);
+                        retries++;
```
Comment (jdnichollsc) [inline src/Proyecto26.RestClient/Helpers/HttpBase.cs:57]: Hola bro! Why did you move these lines?

Comment (fegabe) [reply src/Proyecto26.RestClient/Helpers/HttpBase.cs:57]: Hi! the point of this pull request was to use the retrying callback as a mechanism to repeat a request when failed because of an expired token error. So there is a request that fails because of that, and before retrying again the client notifies of the problem using `RetryCallback` to do the refresh token request and after the delay the client should have had enough time to resolve that request and continue with the original one. That's why I considered moving the callback before the delay

Comment (jdnichollsc) [reply src/Proyecto26.RestClient/Helpers/HttpBase.cs:57]: Oh ok, looks good to me, thanks!

**Review by jdnichollsc — APPROVED** [review]

**Comment by jdnichollsc** [comment]
Thanks for your help mate and sorry for the delay!
